src/WebJobs.Extensions.DurableTask/Options/AzureStorageOptions.cs

```diff
@@ -113,9 +119,37 @@ public void ValidateHubName(string hubName)
             }
             catch (ArgumentException e)
             {
-                throw new ArgumentException(
-                    $"Task hub name '{hubName}' should contain only alphanumeric characters excluding '-' and have length up to 50.", e);
+                throw new ArgumentException(GetTaskHubErrorString(hubName), e);
+            }
+
+            if (hubName.Length > 50)
+            {
+                throw new ArgumentException(GetTaskHubErrorString(hubName));
+            }
+        }
+
+        private static string GetTaskHubErrorString(string hubName)
+        {
+            return $"Task hub name '{hubName}' should contain only alphanumeric characters excluding '-' and have length up to {MaxTaskHubNameSize}.";
+        }
+
+        internal bool IsSanitizedHubName(string hubName, out string sanitizedHubName)
+        {
+            sanitizedHubName = new string(hubName.ToCharArray()
+                                .Where(char.IsLetterOrDigit)
+                                .Take(MaxTaskHubNameSize)
+                                .ToArray());
+            if (sanitizedHubName.Length < MinTaskHubNameSize)
+            {
+                sanitizedHubName = sanitizedHubName + new string(TaskHubPadding, MinTaskHubNameSize - sanitizedHubName.Length);
```

Comment:
Can you summarize what your strategy is here? I was expecting something that simply stripped dashes and perhaps truncated the name.

```csharp
string siteName = "my-site-1";
string hubName = siteName.Replace("-", "");
hubName = hubName.Substring(0, Math.Min(hubName.Length, 45)) + "Hub";
// result is mysite1Hub
```
I'm a bit confused by the padding logic.

Reply:
The padding is for too short of names (i.e. someone has a site-name like like "ab" or "a-b"). I do like your notion of just appending "Hub" instead of just appending `a`'s to the end.